flixel/input/gamepad/ButtonID.hx

```diff
@@ -0,0 +1,57 @@
+package flixel.input.gamepad;
+
+import flixel.system.macros.FlxMacroUtil;
+
+/**
+ * Maps enum values and strings to integer keycodes.
+ */
+@:enum
+abstract ButtonID(Int) from Int to Int
+{
+	public static var fromStringMap(default, null):Map<String, ButtonID>
+		= FlxMacroUtil.buildMap("flixel.input.gamepad.ButtonID");
+		
+	public static var toStringMap(default, null):Map<ButtonID, String>
+		= FlxMacroUtil.buildMap("flixel.input.gamepad.ButtonID", true);
+	
+	// Button Indices
+	var ANY               = -2;		
+	var NONE              = -1;		
+	var A                 =  0;		//BOTTOM face button
+	var B                 =  1;		//RIGHT  face button
+	var X                 =  2;		//LEFT   face button
+	var Y                 =  3;		//TOP    face button
+	var LEFT_SHOULDER     =  4;		//left "bumper"
+	var RIGHT_SHOULDER    =  5;		//right "bumper"
+	var BACK              =  6;		//a.k.a "select", left center button
+	var START             =  7;		//right center button
+	var LEFT_STICK_BTN    =  8;		//Click left stick
```

Comment:
Glad you asked. So, there's a lot of ambiguity involved if we're not careful.

Analog sticks can be clicked, which is a separate digital button input from the movement of the sticks themselves.  Furthermore, on certain gamepads there are additional digital button inputs at the end of the left and right analog shoulder triggers, which means although there's 4 physical shoulder inputs (left/right digital bumpers, left/right analog triggers), there's 6 actual inputs - 4 digital (bumpers + hidden button clicks at end of trigger squeeze), 2 analog (triggers).

So you need to distinguish these parts of the stick:
LEFT_ANALOG_STICK (the stick itself)
LEFT_STICK_BTN (click the stick)

LEFT_TRIGGER (the analog shoulder trigger)
LEFT_TRIGGER_BTN (the digital button at the end of the squeeze on some models)